packages/terra-form-input/src/Input.jsx

```diff
@@ -120,13 +125,18 @@ class Input extends React.Component {
 
     const attributes = { ...customProps };
     const theme = this.context;
+
+    // unsupportedTypes: 'date', 'datetime-local', 'month', 'time', 'week', 'color', 'file', 'range', 'checkbox', 'radio', 'button', 'reset', 'submit'
+    const supportedTypes = [undefined, 'text', 'number', 'password', 'email', 'search', 'tel', 'url', 'hidden'];
```

Comment:
I think what you have here works and we can keep it as is. I just wonder if it would be less confusing to not include `undefined` as a supported type since it is not really an input type. Instead, we can explicitly include the check for `type === undefined` on line 132.

Reply:
+1 - remove `undefined` and instead put it with the check on 132 with something like https://github.com/cerner/terra-core/pull/3386#discussion_r615980437
```suggestion
    const supportedTypes = ['text', 'number', 'password', 'email', 'search', 'tel', 'url', 'hidden'];
```

Reply:
Updated here https://github.com/cerner/terra-core/pull/3386/commits/6ccb549abcd88c1ad6d0af5ea0318a0d38da4652.